src/CommandingOfficers/Commander.java

```diff
@@ -125,12 +126,21 @@ protected void addCommanderAbility(CommanderAbility ca)
   }
 
   /**
-   * Allows a Commander to inject modifications before evaluating a battle.
+   * These functions Allow a Commander to inject modifications before evaluating a battle.
    * Simple damage buffs, etc. can be accomplished via COModifiers, but effects
    * that depend on circumstances that must be evaluated at combat time (e.g. a
    * terrain-based firepower bonus) can be handled here.
+   * applyCombatModifiers() will serve for most combat changes, like the above example.
+   * changeCombatContext() allows the CO to tweak the BattleInstance itself,
+   * to allow for drastic changes to the combat like counterattacking first or at 2+ range.
    */
-  public void applyCombatModifiers(BattleInstance params)
+  public void changeCombatContext(BattleInstance params)
+  {}
+  /**
+   * Allows a Commander to inject modifications that must be evaluated at combat time
+   * @param striking - lets the Commander know if it's dealing or taking damage 
+   */
+  public void applyCombatModifiers(BattleParams params, boolean striking)
```

Comment:
Ehh, I mean, it feels like duplicating information, which is a [cardinal sin in programming](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself); I wouldn't do it that way (Like what if `striking` and the CO check disagree? They never should, but now there's a possibility because there are two places you can check). I won't make you change it if you really want to keep it, since it's in a place that shouldn't change much (ideally), but at least change the variable name to something more informative, like `amITheAttacker` or whatever.

Reply:
...the way it's implemented, it's actually impossible for the pieces of information to disagree (provided COs don't intentionally introduce bugs).
It passes in true to the CO of the unit that's attacking, which means it's basically just doing the exact same check in reverse.
If the system becomes more complicated somehow, then removing it might make sense... but I like how it is at this point.
How about "onOffense"?